sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -520,6 +526,14 @@ case class AdaptiveSparkPlanExec(
           }
       }
 
+    case i: InMemoryTableScanExec =>
```

Comment:
question: if the table cache is already materialized (second access of the cache), do we still need to wrap it with `TableCacheQueryStage`?